### PR TITLE
Validate the versions in pyproject.toml

### PR DESCRIPTION
**Pull request by srittau**
Use packaging.Version instead of a regexp to validate these are valid Python versions.

### CODE REVIEW

**Comment by srittau** [comment]
This reduces the attack vector from a maliciously formed version string in typeshed's pyproject file. It also fixes python/typeshed#12692, where the regexp we used was overly strict.

**Comment by srittau** [comment]
Cc @sobolevn

**Review by sobolevn — APPROVED** [review]
Thank you!

File: tests/test_integration.py [hunk 1]
```diff
@@ -155,9 +155,6 @@ def test_recursive_verify(distribution: str) -> None:
 
 def test_read_typeshed_data() -> None:
     ts_data = read_typeshed_data(Path(TYPESHED))
-    assert re.match(r"^\d+\.\d+\.\d+$", ts_data.mypy_version)
```
Comment (JelleZijlstra) [inline tests/test_integration.py:158]: Is it worth adding a test that we fail if one of these version strings is invalid?

Comment (srittau) [reply tests/test_integration.py:158]: @JelleZijlstra Done. I've now reduced the integration test to just verify that the `read_typeshed_data` function returns and added unit tests for it instead.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**